File: src/atr_pkg/scripts/speed_node.py
#!/usr/bin/env python
# encoding:utf8

import rospy
from std_msgs.msg import String
from qq_msgs.msg import carry
def lidar_callback(msg):{
    rospy.logwarn(msg.data)

}
    
def chao_callback(msg): 
    rospy.loginfo(msg.grade)
    rospy . logwarn(msg.star)
    rospy.loginfo(msg.data)

if __name__ == "__main__":
    rospy.init_node("speed_node")
    sub = rospy.Subscriber("my_chao_topic", carry,  chao_callback, queue_size= 10 )
    sub2 = rospy.Subscriber("my_lidar_topic", carry,  lidar_callback, queue_size= 10 )
    while not rospy.is_shutdown():{
        rospy.spin()
    }

File: src/atr_pkg/src/speed_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
void lidar_callback(std_msgs::String msg)
{
    ROS_INFO(msg.data.c_str());
}

void chao_callback(std_msgs::String msg)
{
    ROS_WARN(msg.data.c_str());
}

int main(int argc, char *argv[])
{

    ros::init(argc, argv, "speed_node");
    setlocale(LC_CTYPE, ""); // 自动继承系统Locale
    
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe<std_msgs::String>("my_lidar_topic", 10, lidar_callback);
    ros::Subscriber sub_2 = nh.subscribe<std_msgs::String>("my_chao_topic", 10, chao_callback);
    while (ros::ok())
    {
        ros::spinOnce();
    }

    return 0;
}